front/components/assistant_builder/server_side_props_helpers.ts

```diff
@@ -33,16 +32,77 @@ import {
   getDefaultTablesQueryActionConfiguration,
   getDefaultWebsearchActionConfiguration,
 } from "@app/components/assistant_builder/types";
+import { getApps } from "@app/lib/api/app";
 import config from "@app/lib/api/config";
+import type { Authenticator } from "@app/lib/auth";
 import { tableKey } from "@app/lib/client/tables_query";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
 import logger from "@app/logger/logger";
 
+export const getAccessibleSourcesAndApps = async (auth: Authenticator) => {
+  const accessibleVaults = await VaultResource.listWorkspaceVaults(
+    auth,
+    !auth.workspace()?.flags.includes("data_vaults_feature") // Will show all vaults if the workspace has the feature flag, otherwise only the global and system vaults
+  );
+
+  const [rawDataSources, dsViews, allDustApps] = await Promise.all([
+    DataSourceResource.listByVaults(auth, accessibleVaults),
+    DataSourceViewResource.listByVaults(auth, accessibleVaults),
+    getApps(auth),
+  ]);
+
+  // Some DataSource have a DataSourceView counterpart, to avoid duplicates in the list of dataSources, we'll remove the ones that have a view.
+  const dataSourceViewIds = dsViews.map((dsView) => dsView.dataSourceId);
+  const dataSourcesWithoutView = rawDataSources.filter(
+    (ds) => !dataSourceViewIds.includes(ds.id)
+  );
+
+  const dataSourcesByName = dataSourcesWithoutView.reduce(
+    (acc, ds) => ({ ...acc, [ds.name]: ds }),
+    {} as Record<string, DataSourceResource>
+  );
+
+  const dataSourceViewsById = dsViews.reduce(
+    (acc, dsView) => ({ ...acc, [dsView.id]: dsView }),
+    {} as Record<string, DataSourceViewResource>
+  );
+
+  {
+    return {
+      dataSources: dataSourcesWithoutView.map((ds) => {
+        return {
+          ...ds.toJSON(),
+          sId: ds.name,
+          vault: ds.vault.toJSON(),
+        };
+      }),
+      dataSourceViews: dsViews.map((dsView) => {
+        return {
+          ...dsView.toJSON(),
+          dataSource: {
+            ...dsView.dataSource!.toJSON(), // dataSource can't be undefined but we had to type it as such because of the way we fetch it
+            sId: dsView.dataSource!.name,
+            vault: dsView.vault.toJSON(),
+          },
+        };
+      }),
+      dataSourcesByName,
+      dataSourceViewsById,
+      dustApps: allDustApps,
+    };
+  }
+};
+
 export async function buildInitialActions({
   dataSourcesByName,
+  dataSourceViewsById,
   dustApps,
   configuration,
 }: {
-  dataSourcesByName: Record<string, DataSourceType>;
+  dataSourcesByName: Record<string, DataSourceResource>;
+  dataSourceViewsById: Record<string, DataSourceViewResource>;
```

Comment:
Could we have only map for both views and regular data sources?

Reply:
I removed the need for passing the maps.
Next update (after endpoint refacto) will only pass one list.